optimizely/odp/odp_manager.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2022, Optimizely
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Optional, Any
+
+from optimizely import exceptions as optimizely_exception
+from optimizely import logger as optimizely_logger
+from optimizely.helpers.enums import Errors, OdpManagerConfig
+from optimizely.helpers.validator import are_odp_data_types_valid
+from optimizely.odp.lru_cache import OptimizelySegmentsCache
+from optimizely.odp.odp_config import OdpConfig, OdpConfigState
+from optimizely.odp.odp_event_manager import OdpEventManager
+from optimizely.odp.odp_segment_manager import OdpSegmentManager
+from optimizely.odp.zaius_graphql_api_manager import ZaiusGraphQLApiManager
+
+
+class OdpManager:
+    """Orchestrates segment manager, event manager and odp config."""
+
+    def __init__(
+        self,
+        disable: bool,
+        segments_cache: OptimizelySegmentsCache,
+        segment_manager: Optional[OdpSegmentManager] = None,
+        event_manager: Optional[OdpEventManager] = None,
+        logger: Optional[optimizely_logger.Logger] = None
+    ) -> None:
+
+        self.enabled = not disable
+        self.odp_config = OdpConfig()
+        self.logger = logger or optimizely_logger.NoOpLogger()
+
+        self.segment_manager = None
+        self.event_manager = None
+
+        if not self.enabled:
+            self.logger.info('ODP is disabled.')
+            return
+
+        if segment_manager:
+            segment_manager.odp_config = self.odp_config
+            self.segment_manager = segment_manager
```

Comment:
what happens if the arguments include a custom cache and a segment manager?

Reply:
Do we support this combo? 
(cache, segman) = (default, default), (custom, default), (-ignored-, custom)

Reply:
Added support for case when arguments include a custom cache and a segment manager. Other cases will be covered by Andy's pr